### PR TITLE
Generate Architectural Decision Records

### PR DESCRIPTION
**Pull request by vfarah-if**
## Description of change

ADR proposal and structure

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"

- [x] Has the branch been rebased to master?
- [x] Automated tests (Any of the following when applicable: Unit, Functional or End-to-End)
- [x] Manual compatibility testing (Browsers: Chrome, Firefox, Edge, Safari)

### CODE REVIEW

File: docs/Architecture/Decisions/0001-record-architecture-decisions.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+## Context
+
+We need to record the architectural decisions made on this project.
+
+## Decision
+
+ - We will use Architecture Decision Records, as [described by Michael Nygard](http://thinkrelevance.com/blog/2011/11/15/documenting-architecture-decisions).
+ - We agreed that any new ADRs should first be raised in a dev huddle before they are put up for general review. This is to ensure that everyone has input on the proposed decision and to ensure that everyone is aware of it.
+
```
Comment (marijnkampf) [inline docs/Architecture/Decisions/0001-record-architecture-decisions.md:17]: Do we also want to add a notification (similar as with a release) to the #data-hub-core-dev slack channel? Not everyone can make it to the dev huddle each week.

Comment (cgsunkel) [reply docs/Architecture/Decisions/0001-record-architecture-decisions.md:17]: I don't think a notification is needed, however when the PR goes up for review it should be made clear in the Slack post that it contains the ADR discussed in a particular meeting

File: docs/Architecture/Decisions/Decisions.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Recorded decisions
+
+```graphviz
```
Comment (Harriethw) [inline docs/Architecture/Decisions/Decisions.md:3]: is this graph written manually, and is the only way to see it through the VS Code preview? Are there any guidelines for its style, e.g. it's just a list of the records with an arrow between them?

Comment (cgsunkel) [reply docs/Architecture/Decisions/Decisions.md:3]: I'm not convinced we need this

File: docs/Architecture/Decisions/0001-record-architecture-decisions.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+## Consequences
+
+See Michael Nygard's article, linked above. For a lightweight ADR toolset, see Nat Pryce's [adr-tools](https://github.com/npryce/adr-tools). Install [Graphviz](https://marketplace.visualstudio.com/items?itemName=geeklearningio.graphviz-markdown-preview) to view generated Graphviz diagrams. Please note this can all be done manually, just follow the format.
+
+```bash
```
Comment (cgsunkel) [inline docs/Architecture/Decisions/0001-record-architecture-decisions.md:22]: I think we should add a `README.md` to the ADR directory that contains these commands, as not everyone will think to open this file to find them.

File: docs/Architecture/Decisions/0002-continue-using-npm-over-yarn.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+## Status
+
+Accepted
+
+## Context
+
+The reason we were considering using Yarn was because NPM initially had a bug with [overrides](https://github.com/npm/cli/issues/4232), that took a long time to resolve within *NPM*, and then *Dependabot* and *Jenkins* internal build packs. 
+
+Afterwards inflating problems with custom versions of the NPM CLI where needed, and the Cloud Foundry issues makeing NPM seem to be the cause for several underlying issues, which in the end tunrned out to be a false positive.
```
Comment (cgsunkel) [inline docs/Architecture/Decisions/0002-continue-using-npm-over-yarn.md:13]:
```suggestion
Afterwards inflating problems with custom versions of the NPM CLI where needed, and the Cloud Foundry issues making NPM seem to be the cause for several underlying issues, which in the end tunrned out to be a false positive.
```

File: docs/Architecture/Decisions/0002-continue-using-npm-over-yarn.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+## Decision
+
+After investingations with the legacy yarn versions and the new bery yarn version, the change seemed to be risky to implement, especially with the impact of all the changes that have been made with Berry.
```
Comment (cgsunkel) [inline docs/Architecture/Decisions/0002-continue-using-npm-over-yarn.md:17]:
```suggestion
After investigations with the legacy yarn versions and the new Berry yarn version, the change seemed to be risky to implement, especially with the impact of all the changes that have been made with Berry. 
```

File: docs/Architecture/Decisions/0002-continue-using-npm-over-yarn.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+The reason we were considering using Yarn was because NPM initially had a bug with [overrides](https://github.com/npm/cli/issues/4232), that took a long time to resolve within *NPM*, and then *Dependabot* and *Jenkins* internal build packs. 
+
+Afterwards inflating problems with custom versions of the NPM CLI where needed, and the Cloud Foundry issues makeing NPM seem to be the cause for several underlying issues, which in the end tunrned out to be a false positive.
+
+## Decision
+
+After investingations with the legacy yarn versions and the new bery yarn version, the change seemed to be risky to implement, especially with the impact of all the changes that have been made with Berry. 
+
+The changes include using its own core mechanism over npm inorder to initialise itself and the project being built. There are two branches as a POC,`feature/add-yarn@1.22.19` and `yarn@berry` that tested the concepts, and if these branches are deleted in the future, search Jira for more information by yarn or the yarn branches mentioned. I am adamant we can get this to work but the effort outweighs the benfit.
```
Comment (cgsunkel) [inline docs/Architecture/Decisions/0002-continue-using-npm-over-yarn.md:19]:
```suggestion
The changes include using its own core mechanism over NPM in order to initialise itself and the project being built. There are two branches as a POC,`feature/add-yarn@1.22.19` and `yarn@berry` that tested the concepts, and if these branches are deleted in the future, search Jira for more information by yarn or the yarn branches mentioned. I am adamant we can get this to work but the effort outweighs the benfit.
```

**Review by cgsunkel — COMMENTED** [review]
There are a few typos that need sorting out and some other changes needed

**Review by marijnkampf — APPROVED** [review]

**Review by Harriethw — APPROVED** [review]

**Review by cgsunkel — APPROVED** [review]

**Pull request merged**